### PR TITLE
Support types On Dashboards Service

### PR DESCRIPTION
**Pull request by simonmacklin**
This PR allows the user to use the available service types to expose opensearch dashboards.  This PR replaces 

```Extend CRD to configure dashboards k8s service #348```

It is the same author but I used the wrong organisation and need to switch to my personal one so I will close the PR above.

### CODE REVIEW

**Comment by simonmacklin** [comment]
Hey @swoehrl-mw 

Do you require any further changes for this PR?

**Review by swoehrl-mw — CHANGES_REQUESTED** [review]
Hi @simonmacklin.
The following small things are missing for me:
* Please add a short paragraph to the userguide.
* Please also copy the changed CRD yaml into the helm chart (see https://github.com/Opster/opensearch-k8s-operator/blob/main/docs/developing.md#submitting-a-pr)

**Review by idanl21 — COMMENTED** [review]
Hey @simonmacklin, first of all thank you for contributing !
second, looks good but we do must add some explanation to the userguide before merging now methods.
Also, like @swoehrl-mw said, we need to add the new CRD Yaml to the helm chart dir to keep it ailing with the repo :) 
please add it and i will merge the PR.

File: docs/userguide/main.md [hunk 1]
```diff
@@ -137,6 +137,64 @@ Using `spec.general.additionalConfig` you can add settings to all nodes, using `
 
 Note that changing any of the `additionalConfig` will trigger a rolling restart of the cluster. If want to avoid that please use the [Cluster Settings API](https://opensearch.org/docs/latest/opensearch/configuration/#update-cluster-settings-using-the-api) to change them at runtime.
 
+## Configuring opensearch_dashboards.yml
```
Comment (swoehrl-mw) [inline docs/userguide/main.md:140]: Looks like there was a hickup with the merging in of the newest main and some paragraphs are now duplicated. Can you please fix that?

**Review by swoehrl-mw — CHANGES_REQUESTED** [review]

**Review by idanl21 — APPROVED** [review]

**Comment by swoehrl-mw** [comment]
@simonmacklin Fixed the userguide myself so we could add your PR to the new release.

**Pull request merged**